Neos.Eel/Classes/Helper/StringHelper.php

```diff
@@ -24,6 +24,43 @@
  */
 class StringHelper implements ProtectedContextAwareInterface
 {
+    /**
+     * Generates a slug of the given string
+     *
+     * Examples::
+     *
+     *     String.urlize('Hello World') == 'hello-world'
+     *     String.urlize('Ä_ÖÜßäöüÅåAÆæØEuro:€éñåøå!_И люяюлб PHP! есть. ﬁ  ') == 'ae-oeuessaeoeueaaaaaaeaeoeuroeenaaoaa-i-luaulb-php-est-fi'
+     *
+     * @param string $string The string to be transformed.
+     * @return string The converted string
+     */
+
+    public function urlize(string $string): string
+    {
+        $specialChars = [
+            // Euro Sign
+            chr(226) . chr(130) . chr(172) => 'E',
+            // GBP (Pound) Sign
+            chr(194) . chr(163) => '',
+            '$' => 'S',
+            'Ä' => 'Ae',
+            'ä' => 'ae',
+            'Ü' => 'Ue',
+            'ü' => 'ue',
+            'Ö' => 'Oe',
+            'ö' => 'oe',
+            // Norwegian characters
+            'Å' => 'Aa',
+            'å' => 'aa'
+        ];
+        $string = strtr($string, $specialChars);
+        $string = str_replace('_', ' ', $string);
+        $string = transliterator_transliterate('Any-Latin; Latin-ASCII; [\u0100-\u7fff] remove; [:Punctuation:] Remove; Lower()', $string);
```

Comment:
Oh, nice!
Just wondering: is there a specific reason to use procedural style here?
As a little performance optimization I would suggest to go for sth like:

```php
    /**
     * @var \Transliterator
     */
    private static $transliterator;

    private function getTransliterator(): \Transliterator
    {
        if (self::$transliterator === null) {
            self::$transliterator = \Transliterator::createFromRules('...');
        }
        return self::$transliterator;
    }

    // ...

    public function urlize(string $string): string
    {
        $string = $this->getTransliterator()->transliterate($string);
        // ...
```